04.ls/ls_command.rb

```diff
@@ -10,11 +10,13 @@ def main
 
   options = {}
   opt.on('-a') { |v| options[:a] = v }
+  opt.on('-r') { |v| options[:r] = v }
   opt.parse(ARGV)
 
   all_files = find_all_files
   visible_files = options[:a] ? all_files : select_visible_files(all_files)
-  display_files(visible_files)
+  reverse_visible_files = visible_files.reverse if options[:r]
+  display_files(reverse_visible_files || visible_files)
```

Comment:
ここのifと次の行の`||`で2回条件分岐が発生するので、

```suggestion
  hoge = options[:r] ? visible_files.reverse : visible_files
  display_files(hoge)
```

のようにした方が脳に優しいです。
ただし、変数名を考えるのはちょっと難しいかも？いい名前を考えてみてください。